client.lua

```diff
@@ -321,50 +359,32 @@ function Init()
                 SetEntityHeading(veh, Config.Shops[k]["ShowroomVehicles"][i].coords.w)
                 FreezeEntityPosition(veh, true)
                 SetVehicleNumberPlateText(veh, 'BUY ME')
-                if Config.UsingTarget then createVehZones(k, veh) end
+                if Config.UsingTarget then createVehicleZones(k, veh) end
             end
-            if not Config.UsingTarget then createVehZones(k) end
+            if not Config.UsingTarget then createVehicleZones(k) end
         end
     end)
 end
 
--- Events
+--- Opens the vehicleshop menu
 RegisterNetEvent('qb-vehicleshop:client:homeMenu', function()
     openVehicleSellMenu()
 end)
 
-RegisterNetEvent('qb-vehicleshop:client:TestDrive', function()
-    if not inTestDrive and ClosestVehicle ~= 0 then
-        inTestDrive = true
-        tempShop = insideShop -- temp hacky way of setting the shop because it changes after the callback has returned since you are outside the zone
-        QBCore.Functions.TriggerCallback('QBCore:Server:SpawnVehicle', function(netId)
-            local veh = NetToVeh(netId)
-            exports['LegacyFuel']:SetFuel(veh, 100)
-            SetVehicleNumberPlateText(veh, 'TESTDRIVE')
-            SetEntityHeading(veh, Config.Shops[tempShop]["TestDriveSpawn"].w)
-            TriggerEvent('vehiclekeys:client:SetOwner', QBCore.Functions.GetPlate(veh))
-            testDriveVeh = netId
-            lib.notify({
-                title = Lang:t('general.testdrive_timenoti',
-                    { testdrivetime = Config.Shops[tempShop]["TestDriveTimeLimit"] }),
-                type = 'inform'
-            })
-        end, Config.Shops[tempShop]["ShowroomVehicles"][ClosestVehicle].chosenVehicle,
-            Config.Shops[tempShop]["TestDriveSpawn"], true)
-        startTestDriveTimer(Config.Shops[tempShop]["TestDriveTimeLimit"] * 60, tempShop)
-    else
-        lib.notify({
-            title = Lang:t('error.testdrive_alreadyin'),
-            type = 'error'
-        })
-    end
-end)
-
-RegisterNetEvent('qb-vehicleshop:client:customTestDrive', function(data)
+--- Starts the test drive. If vehicle parameter is not provided then the test drive will start with the closest vehicle to the player.
+--- @param vehicle number | nil
+RegisterNetEvent('qb-vehicleshop:client:testDrive', function(vehicle)
     if not inTestDrive then
+        local testDriveVehicle
         inTestDrive = true
-        local vehicle = data
-        tempShop = insideShop -- temp hacky way of setting the shop because it changes after the callback has returned since you are outside the zone
+        tempShop = insideShop
+
+        if vehicle then
+            testDriveVehicle = vehicle
+        else
+            testDriveVehicle = Config.Shops[tempShop]["ShowroomVehicles"][ClosestVehicle].chosenVehicle
+        end
```

Comment:
Can define testDriveVehicle here using a ternary https://manason.github.io/effective-fivem-lua/conditionals/#default-values

Also https://manason.github.io/effective-fivem-lua/tables/#prefer-object-access-for-constant-keys-and-array-access-for-non-constant-keys